base/server/upgrade/11.9.0/03-UpdateMLDSAProfiles.py

```diff
@@ -0,0 +1,94 @@
+# Authors:
+#     Marco Fargetta <mfargett@redhat.com>
+#
+# Copyright Red Hat, Inc.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import logging
+import os
+
+import pki
+
+logger = logging.getLogger(__name__)
+
+#NOTE: the update does not work with LDAP stored profiles
+class UpdateMLDSAProfiles(pki.server.upgrade.PKIServerUpgradeScriptlet):
+
+    def __init__(self):
+        super().__init__()
+        self.message = 'Update CA profiles for ML-DSA support'
+
+    def upgrade_subsystem(self, instance, subsystem):
+
+        if subsystem.name != 'ca':
+            return
+
+        # New ML-DSA profiles to add
+        new_profiles = [
+            'caMLDSAUserCert',
+            'caMLDSAServerCert',
+            'caMLDSASubsystemCert',
+            'caMLDSAAdminCert',
+        ]
+
+        # Read the available profile list
+        profile_list = subsystem.config.get('profile.list').split(',')
+
+        # Update all profiles
+        # If the signature algorithms have been modified in the instance
+        # the new algorithms are not included
+        new_signing_algs = {'ML-DSA-87', 'ML-DSA-44', 'ML-DSA-65'}
+        for profile in profile_list:
+            file_name = '{}.cfg'.format(profile)
+            new_profile = {}
+            instance_profile = {}
+            pki.util.load_properties(
+                '/usr/share/pki/ca/profiles/ca/{}'.format(file_name),
+                new_profile)
+            path = os.path.join(subsystem.base_dir, 'profiles', 'ca', file_name)
+            if not os.path.exists(path):
+                continue
+
+            pki.util.load_properties(
+                path,
+                instance_profile)
+            keys = [key for key, val in instance_profile.items() if 'signingAlgsAllowed' in key]
```

Comment:
Should we use a regex or a longer keyword (e.g. `.constraint.params.signingAlgsAllowed`) to avoid accidental match? If it's considered safe enough no need to modify.